### PR TITLE
Simplify indentStack code

### PR DESCRIPTION
**Pull request by stefan-koch-sociomantic**
as per title, this PR makes some code in the IndentStack structure clearer.

### CODE REVIEW

File: src/dfmt/indentation.d [hunk 1]
```diff
@@ -67,16 +67,21 @@ struct IndentStack
...
     void push(IdType item) pure nothrow
     {
-        arr[index] = item;
-        index = index + 1 == arr.length ? index : index + 1;
+        arr[index++] = item;
+        if (index == arr.length)
+        {
+            //FIXME this is actually a bad thing to do
+            index--;
+        }
```
Comment (don-clugston-sociomantic) [inline src/dfmt/indentation.d]: I think you should just do ` arr[index]=item;  if ( index < arr.length ) ++index; `

Comment (stefan-koch-sociomantic) [reply src/dfmt/indentation.d]: That's  less brittle, yes.
But I'd like it to stay brittle such that I have more arguments in favour for changing how this is done.

Comment (stefan-koch-sociomantic) [reply src/dfmt/indentation.d]: oops this can be read by everyone ....

Comment (Hackerpilot) [reply src/dfmt/indentation.d]: It can.

**Comment by stefan-koch-sociomantic** [comment]
@Hackerpilot I changed to the less brittle approach please merge.

**Comment by leandro-lucarella-sociomantic** [comment]
Is this a bug fix? If it is, please mark it as such with the labels. If it's not, it should go to v0.x.x.

**Comment by don-clugston-sociomantic** [comment]
LGTM

**Pull request merged**